### ISSUE TITLE
Potential false positive with `UtcDateTime.fromTimestamp` in CLJS

### ISSUE THREAD

**Issue by p-himik** [borkdude/clj-kondo#556]
**version**

2019.10.26

**platform**

Ubuntu 19.10

**problem**

Many projects, including one of mines, use `UtcDateTime.fromTimestamp` in just that exact form, with an addition of a prior `(:import [goog.date UtcDateTime])`.
A real world example: https://github.com/andrewmcveigh/cljs-time/blob/ea76ee787353762088892912f8c17863316e9f6d/src/cljs_time/coerce.cljs#L29

**repro**

```bash
$ git clone https://github.com/andrewmcveigh/cljs-time.git && cd cljs-time
$ clj-kondo --lint src/cljs_time/coerce.cljs | grep fromTimestamp
src/cljs_time/coerce.cljs:29:18: error: unresolved symbol UtcDateTime.fromTimestamp
```

The linter could be made happy by using `(:require [goog.date.UtcDateTime :as udt])` and using `udt/fromTimestamp` instead.

**Comment by borkdude** [comment]
Repro:

example.cljs:

``` clojure
(ns foo (:import [goog.date UtcDateTime]))
(defn utc [x] (UtcDateTime.fromTimestamp x))'
```
```
$ clj-kondo --lint example.cljs
<stdin>:1:59: error: unresolved symbol UtcDateTime.fromTimestamp
linting took 36ms, errors: 1, warnings: 0
```

**Comment by borkdude** [comment]
Recorded from a Slack conversation in #clojurescript:

```
borkdude  I was fixing a false positive in clj-kondo (a clojure linter) because people seem to be using it like this:
(ns foo (:import [goog.date UtcDateTime]))
(defn utc [x] (UtcDateTime.fromTimestamp x))
Maybe they shouldn't, but it has been seen in a couple of code bases

is UtcDateTime a class? I don't know if this is even a concept in JS (outside ES6, TS, etc.), hence my question
they seem to be calling it like that over here: https://google.github.io/closure-library/api/goog.date.UtcDateTime.html
dnolen  that's idiomatic
goog.date.UtcDateTime is a library and a ctor
thheller  UtcDateTime is a class with static methods
dnolen  ok, but that doesn't matter so much - :require won't let you do what you want
borkdude  that's informative, thanks (edited) 
dnolen  the point is that the thing you want to use and the namespace are the same
(:import [foo.bar Baz]) lets you require a namespace called foo.bar.Baz & use Baz as if referred
borkdude  according to the issuer you can also:
(:require [goog.date.UtcDateTime :as udt])
udt/fromTimestamp
dnolen  right
sorry if it's a class with only static methods and you don't need to use it as a ctor
then :require is fine
```
